### PR TITLE
PBs: Update Playbooks to conform with ansible lint v4.2

### PR DESCRIPTION
**Pull request by Willsparker**
@sxa555 @karianna 

Ref: #783 #782 

First batch of changes required to make our playbooks conform to `ansible-lint 4.2.0`.

### CODE REVIEW

**Comment by Willsparker** [comment]
Now that the PR is out of WIP, I thought I'd explain the main bulk of changes:
- replacing `x == false` to `x == "false"`, seems to just be a best practise thing.
- placing the `skip_ansible_lint` tag on any task that I don't think I could fix the linter's issue with. The bulk of these is checking the versions for things and the linter throwing the `Use shell only when shell functionality is required` error,
- setting `pipefail` when shell modules are using pipes
- Replacing shell tasks to use modules that are better suited, i.e. using `lineinfile` module for `sed` commands, using `alternatives` module for `alternatives` commands, using `locale_gen` module for `localedef` commands

**Comment by Willsparker** [comment]
`vagrantPlaybookCheck` job failed, putting into [wip] until I can get a run of it succeeding

**Comment by karianna** [comment]
@Willsparker needs a rebase.

**Comment by karianna** [comment]
@Willsparker - This looks done now?  Shall I review?

**Comment by Willsparker** [comment]
Please :-) I have got a 'mostly' working run on my Jenkins job, any fallout I'll look into correcting this week - I'm currently in the process of fixing a git_source issue anyway

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/RedHat.yml [hunk 1]
```diff
@@ -144,56 +146,72 @@
...
-  shell: update-alternatives --set java java-1.8.0-openjdk.aarch64
+  alternatives:
+    name: java
+    path: java-1.8.0-openjdk.aarch64
   when:
     - ansible_distribution_major_version == "7"
     - ansible_architecture == "aarch64"
   tags: default_java
 
 - name: Set default java version - RHEL 7 ppc64le
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/RedHat.yml]: Incorrect naem, should read s390?

Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/RedHat.yml]: Good spot

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/RedHat.yml [hunk 2]
```diff
@@ -144,56 +146,72 @@
...
   when:
     - ansible_distribution_major_version == "6"
     - ansible_architecture == "ppc64"
   tags: default_java
 
 - name: Set default java version - RHEL 7 ppc64
-  shell: update-alternatives --set java /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.161-0.b14.el7_4.ppc64/jre/bin/java
+  alternatives:
+    name: java
+    path: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.161-0.b14.el7_4.ppc64/jre/bin/java
```
[the remark this answers is not in the document]
Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/RedHat.yml]: Possibly  - I'm not really able to check at this moment in time as I don't have access to a ppc64le machine though. That being said, there is `java-1.8.0-openjdk-1.8.0.222.b03-1.el7.ppc64le.rpm` in the CentOS7 repository: http://mirror.centos.org/altarch/7/os/ppc64le/Packages/
Not sure whether this is very accurate to RHEL7's repo though..

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/CentOS.yml [hunk 3]
```diff
@@ -192,24 +222,34 @@
   shell: locale -a
   register: localeList
   changed_when: False
-  tags: locales
+  tags:
+    - locales
+    - skip_ansible_lint
```
[the remark this answers is not in the document]
Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/CentOS.yml]: Fixed 👍

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml [hunk 4]
```diff
@@ -30,7 +30,13 @@
   tags: patch_update
 
 - name: Change Jessie to Trusty in the sources files
-  command: sed -i "s/jessie/trusty/g" /etc/apt/sources.list.d/ppa_openjdk_r_ppa_jessie.list  /etc/apt/sources.list.d/ppa_ubuntu_toolchain_r_test_jessie.list
+  replace:
+    path: "{{ item }}"
```
[the remark this answers is not in the document]
Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Debian.yml]: fixed 👍

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/FreeBSD.yml [hunk 5]
```diff
@@ -9,6 +9,8 @@
 - name: Get number of cores on FreeBSD
   shell: sysctl hw.ncpu | awk '{print $2}'
   register: freebsd_cores
+  tags:
+    - skip_ansible_lint
```
[the remark this answers is not in the document]
Comment (Willsparker) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/FreeBSD.yml]: Done for all addition of `skip_ansible_lint` tags :-)

**Pull request merged**